src/BuildQueue.cpp

```diff
@@ -97,8 +97,6 @@ void BuildQueueInit(BuildQueue *queue, const BuildQueueConfig *config, const cha
 
     Log(kDebug, "build queue initialized; ring buffer capacity = %u", queue->m_QueueCapacity);
 
-    // Block all signals on the main thread.
```

Comment:
Do we know why the signals were being blocked here previously?

Reply:
Yes. The old signal handling code relied on:
1. Calling sigwait on a signal handling thread
2. Blocking signals on the main thread, so it does not get killed when a signal comes in.

We don't know why this approach was chosen over a more simple signal handler callback approach (With a signal handler, we don't need to block signals). But we did find that it did not work on linux, and this does.